### ISSUE TITLE
Extracted flow 0.53 props from react component are not recognized

### ISSUE THREAD

**Issue by danrot** [reactjs/react-docgen#207]
I have just changed our codebase to the new flow types for React (meaning to use stuff as described [here](https://flow.org/en/docs/react/components/#toc-class-components)).

So my react classes now look like that:

```javascript
type Props = {
    value: boolean,
};

class MyComponent extends React.Component<Props> {
    // ...
}
```

These props are then not recognized by the library anymore... I am using version 2.17.0.

**Comment by rosskevin** [comment]
Please change the title to something referencing Flow 53, hopefully this will get everyone's attention

**Comment by danrot** [comment]
@rosskevin Changed it, but I doubt that this helps...

**Comment by danez** [comment]
Feel free to make a PR.

We should also make it work also with old and new flow versions

something like

```js
const params = superTypes.get('params');
if (params.value.length === 3) {
  typePath = params.get(1);
} else {
 typePath = params.get(0);
}
```

**Comment by danez** [comment]
I created a PR which fixes this #209
